src/api/argo-templates/base/sat-general-template-iuf.argo.yaml

```diff
@@ -0,0 +1,151 @@
+              container.
+          - name: script_stdout_file
+            default: "/tmp/script_stdout_file"
+          - name: script_content
+            description: >
+              The script that will be executed after the sat CLI is configured. The stdout
+              of this script will be captured and returned as an output parameter from this
+              template.
+      outputs:
+        parameters:
+          - name: script_stdout
+            valueFrom:
+              path: "{{inputs.parameters.script_stdout_file}}"
+      script:
+        image: artifactory.algol60.net/sat-docker/stable/cray-sat:3.20.0
+        command: [sh]
+        source: |
+          #!/bin/sh
+          ts="$(mktemp)"
+          cat <<'SAT_GENERAL_EOF' > "${ts}"
+          function setup() {
+            set -e
+
+            # This reads from /usr/local/share/ca-certificates and updates /etc/ssl/certs/ca-certificates.crt
+            update-ca-certificates 2>/dev/null
+
+            sat init
+
+            # Write the auth token parameter to the file
+            JSON_FMT='{"access_token":"%s"}\n'
+            printf "$JSON_FMT" "{{inputs.parameters.auth_token}}" > /root/.config/sat/token.json
+            echo -n $(tr -d "\n\r" < /root/.config/sat/token.json) > /root/.config/sat/token.json
+
+            # Set up the SAT config to use the token file
+            sed -i -e 's|.*token_file.*|token_file = "/root/.config/sat/token.json"|' /root/.config/sat/sat.toml
+
+            # TODO: Fix sat bootprep code to load kube config from cluster
+            # To workaround this issue for now, construct kubeconfig file in the pod
+            secret=`kubectl -n argo get serviceaccount/default -o jsonpath='{.secrets[0].name}'`
+            token=`kubectl -n argo get secret $secret -o jsonpath='{.data.token}'| base64 -d`
+            mkdir -p mykubeconfig
+            cat << EOF > mykubeconfig/admin.conf
+            apiVersion: v1
+            kind: Config
+            current-context: default
+            contexts:
+              - context:
+                  cluster: kubernetes
+                  user: default
+                  namespace: default
+                name: default
+            clusters:
+              - cluster:
+                  server: https://kubeapi-vip.local:6442
+                  insecure-skip-tls-verify: true
+                name: kubernetes
+            users:
+            - name: default
+              user:
+                token: ${token}
+          EOF
+            export KUBECONFIG=mykubeconfig/admin.conf
+
+            # TODO: This is required to build images in IMS with sat bootprep. Find a way to pass
+            # an IMS public key id into the argo workflow. Perhaps from the IUF CLI.
+            ssh-keygen -f ~/.ssh/id_rsa -N ''
+            set +e
+          }
+
+          function execute_script_content() {
+            {{inputs.parameters.script_content}}
```

Comment:
If there is a closing brace at the beginning of script content, arbitrary code can be executed before `main`.

Reply:
Jack and I discussed and decided that it's most likely not a security concern if they are able to do this as the function executes in the same context as the code outside the function anyway. It is just something to be aware of for callers of this shared template.